Content.Server/Chemistry/EntitySystems/InjectorSystem.cs

```diff
@@ -330,8 +330,21 @@ private void AfterDraw(Entity<InjectorComponent> injector, EntityUid target)
             return;
         }
 
+        var applicableTargetSolution = targetSolution.Comp.Solution;
+        // If a whitelist exists, remove all non-whitelisted reagents from the target solution temporarily
+        Solution temporarilyRemovedSolution = new();
+        if (injector.Comp.ReagentWhitelist is { } reagentWhitelist)
+        {
+            string[] reagentPrototypeWhitelistArray = new string[reagentWhitelist.Count];
+            for (var i = 0; i < reagentWhitelist.Count; ++i)
+            {
+                reagentPrototypeWhitelistArray[i] = reagentWhitelist[i];
+            }
```

Comment:
```suggestion
            var reagentPrototypeWhitelistArray = reagentWhitelist.Select(x => (string)x).ToArray();
```

Reply:
note: If not for the fact C# does require the name to be different, I would be recommending the name be shortened to not include it's type (as it's redundant information), as well.